webapp/src/js/app.js

```diff
@@ -109,6 +109,10 @@ _.templateSettings = {
     remote: {
       skip_setup: true,
       fetch: function(url, opts) {
+        // Hack to stop service worker intefere with CouchDB replication
+        if (!url.split('/')[3]) {
+          url = `${url}api/info`;
+        }
```

Comment:
What is this doing? Why does the service worker interfere?

Reply:
I still don't get this. Why are we changing the URL to `/api/info`?

Reply:
`PouchDB` replication throws error, from my understanding pouch replication only works with `documents` and `blob`. by the time the fetch url is `http://localhost:5988/` the url returns a cached service worker page i.e. `http://localhost:5988/?_sw-precache....` this is a `text/html` content and pouch fails to replicate

I tried setting the Url to fetch the original document i.e. `http://localhost:5988/medic/_design/medic/templates/inbox.html` this throws 403 error, and if pouch replication encounters 403 it aborts

I don't understand Pouch Replication  to details because also the error message looks scanty

Reply:
Why is the replication requesting `/`? Didn't we set the remote db to `/medic`? I wonder if it's trying to get the couchdb info...

Reply:
Yes the remoteDB is set to `/medic`, it has always been requesting `/` even on the main branch I tested that, maybe you could checkout the branch and see since I may be lacking all the pieces to complete this puzzle

Reply:
Ok I just had a quick look and I think pouch is trying to get info about the couch server. Without digging through the pouchdb code I'm not sure why.

It's annoying that the service worker is getting in the way...

I think your solution is ok but we should make sure the response is what pouchdb expects, so can we forward that request on to couchdb rather than returning api info?

Reply:
Yes exactly it is trying to get the couchdb info, now the challenge is the race condition  that SW has put us on with regards to `/` maybe I will try and add _ignoreUrlParametersMatching [Array⟨Regex⟩]_ to service-worker then request `/?couch_request...` so service worker backs off

Reply:
Sure. Or create a new API for `/couch/info` or something, which proxies on to couch and gets the real info. But yeah, anything to get around the SW!